### ISSUE TITLE
Improve faulty storage media techniques

### ISSUE THREAD

**Issue by 20k-ultra** [balena-io/docs#1614]
We have https://www.balena.io/docs/learn/more/masterclasses/device-debugging/#112-storage-media-corruption and https://www.balena.io/docs/faq/troubleshooting/troubleshooting/#sd-card-corruption but both do not do a good job helping people know how to identify faulty SD cards or other storage medias. 

If you run healthchecks you get errors such as:

Some localdisk issues detected: 
test_write_latency Slow disk writes detected: mmcblk0p1: 3260.62ms / write, sample size 65
mmcblk0p5: 16837.2ms / write, sample size 1623
test_data_partition_mounted Data partition not mounted read-write

I feel this information should be added to these docs to explain why these are indicators of faulty storage media.

**Comment by 20k-ultra** [comment]
another point to add is that the device runs `fsck` on boot so performing a reboot will trigger it to run... at least that's what I recall but this is another example of documentation we can improve on.

**Comment by xginn8** [comment]
we already document those check results in their respective docs, e.g. https://www.balena.io/docs/reference/diagnostics/#test_data_partition_mounted

**Comment by 20k-ultra** [comment]
Yup, I know about those but our corrupt storage media docs are pretty lacking in things to troubleshoot or info on what to probe so I want to move that info into here. Someone won't get to the device healthchecks when they search "faulty sd card"

**Comment by jakogut** [comment]
In researching the same problem, I came across [badblocks](https://wiki.archlinux.org/index.php/badblocks), a program for writing patterns of bytes to disk, and detecting errors, a la Memtest86 for block devices.

**Comment by 20k-ultra** [comment]
https://www.balena.io/docs/learn/more/masterclasses/device-debugging/#112-storage-media-corruption should be updated to mention that /etc/resin-supervisor/supervisor.conf will always fail the fingerprint test when a user upgrades the Supervisor. This file tracks the Supervisor version configured to run on a device so if a user updates this then it will not match the fingerprint.

**Comment by 20k-ultra** [comment]
Checking kernel logs for FS errors with something like `dmesg | grep -i 'ext'` is a good trick. We can also advise unmounting the target partition that is unhealthy and run fsck on it.
